core/src/main/java/com/oath/oak/OakRReference.java

```diff
@@ -25,7 +26,7 @@
  * before each access since it can only be used without other concurrent writes in the background.
  * */
 
-public class OakRReference implements OakRBuffer {
+public class OakRReference implements OakRBuffer, OakUnsafeRef {
```

Comment:
Need to find a better name for this class, like OakReadOnlyBuffer maybe?

Reply:
I agree. I suggest to group many renaming like this and do a different PR for renaming ONLY.

Reply:
I agree that this name is bad. I think it should be OakRKeyBufferImpl (or similar). I also think that naming of the Oak output buffers only can be held under this PR.

Reply:
Especially not to add the NEW Reference name...